kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
+
+      -- user wanted a list of all active targets for this upstream
+      local target_history, err = dao_factory.targets:find_all({
+        upstream_id = self.params.upstream_id,
+      })
+
+      if not target_history then
+        return app_helpers.yield_error(err)
+      end
+
+      -- sort and walk based on target and creation time
+      for _, target in ipairs(target_history) do
+        target.order = target.target..":"..target.created_at
+      end
+      table.sort(target_history, function(a, b) return a.order > b.order end)
```

Comment:
Thing is, your examples **_are_** identical. And there is nothing we can do about it. `timestamp == timestamp`. We simply do not have any information on which one comes first. Increasing timestamp precision would only make it less likely for a collision to happen, but won't take the possibility away.

That given: all we can do is make sure that whenever we rely on the order of events, we make sure to use the same order. The uuid is created with the record, and never changes afterwards, so it is a perfect way to make the sort order deterministic. So yes, the order is arbitrary in the end, but it will be consistently the same order on every occasion.